sweetalert2.d.ts

```diff
@@ -336,53 +336,13 @@ declare module 'sweetalert2' {
      */
     const version: string
   }
-
-  export type SweetAlertIcon = 'success' | 'error' | 'warning' | 'info' | 'question';
-
-  export type SweetAlertInput =
-    'text' | 'email' | 'password' | 'number' | 'tel' | 'range' | 'textarea' | 'select' | 'radio' | 'checkbox' |
-    'file' | 'url';
-
-  export type SweetAlertPosition =
-    'top' | 'top-start' | 'top-end' | 'top-left' | 'top-right' |
-    'center' | 'center-start' | 'center-end' | 'center-left' | 'center-right' |
-    'bottom' | 'bottom-start' | 'bottom-end' | 'bottom-left' | 'bottom-right';
-
-  export type SweetAlertGrow = 'row' | 'column' | 'fullscreen' | false;
-
-  export interface SweetAlertResult<T = any> {
-    value?: T;
-    dismiss?: Swal.DismissReason;
-    isConfirmed: boolean;
-    isDismissed: boolean;
-  }
-
-  export interface SweetAlertShowClass {
-    popup?: string;
-    backdrop?: string;
+  interface SweetAlertCommonClass {
     icon?: string;
-  }
-
-  export interface SweetAlertHideClass {
     popup?: string;
-    backdrop?: string;
-    icon?: string;
   }
 
-  export interface SweetAlertCustomClass {
-    container?: string;
-    popup?: string;
-    header?: string;
-    title?: string;
-    closeButton?: string;
-    icon?: string;
-    image?: string;
-    content?: string;
-    input?: string;
-    actions?: string;
-    confirmButton?: string;
-    cancelButton?: string;
-    footer?: string;
+  interface SweetAlertHideShowClass extends SweetAlertCommonClass {
```

Comment:
`SweetAlertCommonClass` is an over-kill here, please remove it as it makes code less readable.

Reply:
> `SweetAlertCommonClass` is an over-kill here, please remove it as it makes code less readable.

Hmm, ok... but did you notice that I used it in `SweetAlertHideShowClass` and `SweetAlertCustomClass`?

Reply:
Yes, https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
Thanks for the link :) I sent a fixup for that now.